src/Gelf4Net.HttpAppender/GelfHttpAppender.cs

```diff
@@ -42,14 +42,23 @@ public override void ActivateOptions()
         protected override void Append(LoggingEvent loggingEvent)
         {
             var payload = RenderLoggingEvent(loggingEvent);
-            var content = new StringContent(payload, Encoding.UTF8, "application/json");
-            _httpClient.PostAsync(_baseUrl, content).ContinueWith(CallBackAfterPost);
+            var _ = SendMessageAsync(payload);
         }
 
-        private void CallBackAfterPost(Task<HttpResponseMessage> obj)
+        protected async Task SendMessageAsync(string payload)
         {
-            if (obj.Exception != null)
-              ErrorHandler.Error("Unable to send logging event to remote host " + this.Url, obj.Exception);
+            try
+            {
+                using (var content = new ByteArrayContent(payload.GzipMessage(Encoding.UTF8)))
+                {
+                    content.Headers.ContentEncoding.Add("gzip");
+                    await _httpClient.PostAsync(_baseUrl, content);
```

Comment:
Yes, otherwise:

 * The `content` could be disposed before the task continues
 * Some exceptions might not be caught by the handler below.

For the 'fire and forget' style of the basic `GelfHttpAppender` the result of this is discarded
without an `await` in the `Append` method.